variables.tf

```
@@ -288,8 +288,21 @@ variable "protect_from_deletion" {
 
 variable "space_id" {
   type        = string
-  description = "Place the created stacks in the specified space_id."
-  default     = "root"
+  description = "Place the created stacks in the specified space_id. Mutually exclusive with space_name."
+  default     = null
+}
+
+variable "space_name" {
+  type        = string
+  description = "Place the created stacks in the specified space_name. Mutually exclusive with space_id."
+  default     = null
+}
+
+check "spaces_enforce_mutual_exclusivity" {
+  assert {
+    condition     = var.space_id == null || var.space_name == null
+    error_message = "space_id and space_name is mutually exclusive."
+  }
 }
```

Few things here: 

1. Hm... where do checks live... I wouldn't expect to see this here, but that calls in to question: Does they live in main or do they have their own file maybe? Let's move it to `main.tf` below the `locals` block. 
2. One correction on grammar
3. Shouldn't the logic be `space_id != null && space_name != null`?

```suggestion
check "spaces_enforce_mutual_exclusivity" {
  assert {
    condition     = var.space_id == null || var.space_name == null
    error_message = "space_id and space_name are mutually exclusive."
  }
}
```

---

1. I thought of it in the `variables.tf` so it's close and right near the variables. But it does feel weird being there... Moved to main.tf
2. Thanks 👍 
3. It's mutually exclusive but doesn't have to be exhaustive. If `space_id != null && space_name != null`, it forces one of these to have a value // forces both to not be null.
  - We only one want of these to have a value, or let both be optional default

---

The default null falls back to "root". If we set space_id default as "root" in the variables, then it becomes messy when someone wants to use var.space_name, because they will have to manually set space_id as null.

e.g.
```hcl
module "spacelift_automation" {
  source = "...github.com/masterpointio/terraform-spacelift-automation..."

  # Avoid this
  space_name = "production"
  space_id = null # defaulted to "root"
}
```

---

Re-review on this? Addressed

---

I think I may be brain dead today... the amount of time I just spent to think about this simple logic is weird. 

Anyway, I think you're right. Though this does seem overly complex... Let's :shipit: